src/mongo/db/redaction_parameter.cpp

```diff
@@ -0,0 +1,85 @@
+/*======
+This file is part of Percona Server for MongoDB.
+
+Copyright (c) 2006, 2017, Percona and/or its affiliates. All rights reserved.
+
+    Percona Server for MongoDB is free software: you can redistribute
+    it and/or modify it under the terms of the GNU Affero General
+    Public License, version 3, as published by the Free Software
+    Foundation.
+
+    Percona Server for MongoDB is distributed in the hope that it will
+    be useful, but WITHOUT ANY WARRANTY; without even the implied
+    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+    See the GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public
+    License along with Percona Server for MongoDB.  If not, see
+    <http://www.gnu.org/licenses/>.
+======= */
+
+#define MONGO_LOG_DEFAULT_COMPONENT ::mongo::logger::LogComponent::kDefault
+
+#include "mongo/base/init.h"
+#include "mongo/db/server_options.h"
+#include "mongo/db/server_parameters.h"
+#include "mongo/util/log.h"
+
+namespace mongo {
+
+// Allow --redactClientLogData access via setParameter/getParameter
+namespace {
+
+class RedactClientLogDataParameter : public ServerParameter
+{
+    MONGO_DISALLOW_COPYING(RedactClientLogDataParameter);
+
+public:
+    RedactClientLogDataParameter(ServerParameterSet* sps);
+    virtual void append(OperationContext* txn, BSONObjBuilder& b, const std::string& name);
+    virtual Status set(const BSONElement& newValueElement);
+    virtual Status setFromString(const std::string& str);
+
+private:
+    Status _set(bool v);
+};
+
+MONGO_INITIALIZER_GENERAL(InitRedactClientLogDataParameter,
+                          MONGO_NO_PREREQUISITES,
+                          ("BeginStartupOptionParsing"))(InitializerContext*) {
+    new RedactClientLogDataParameter(ServerParameterSet::getGlobal());
```

Comment:
Can you please implement the parameter as static variable like here: https://github.com/percona/percona-server-mongodb/blob/master/src/mongo/db/commands/parameters.cpp#L279 ?